tensorflow/c/experimental/filesystem/plugins/gcs/BUILD

```diff
@@ -0,0 +1,33 @@
+# Experimental GCS filesystem plugin.
+load("//tensorflow:tensorflow.bzl", "tf_cc_shared_object")
+
+package(
+    default_visibility = ["//visibility:private"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+# Filesystem implementation for GCS environments
+tf_cc_shared_object(
+    name = "gcs_filesystem",
+    framework_so = [],
+    linkstatic = False,
+    visibility = ["//visibility:public"],
+    deps = [":gcs_filesystem_impl"],
+    per_os_targets = 1,
+)
+
+# The real implementation of the filesystem.
+cc_library(
+    name = "gcs_filesystem_impl",
+    srcs = ["gcs_filesystem.cc"],
+    copts = select({
+        "//conditions:default": ["-std=c++17"],
+        "//tensorflow:macos": ["-lc++fs", "-std=c++17"],
+        "//tensorflow:windows": ["/std:c++17"],
```

Comment:
I think we can use C++ 17 here because these copts are limited to the plugins only. I've already compiled these code on windows and it runs extracly on linux